File: backend-java/src/main/java/com/normanblog/frestyle/entity/LearningReport.java
package com.normanblog.frestyle.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.time.Instant;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/** ユーザーの学習レポート(週次・月次集計を非同期で生成)を表す JPA エンティティ。 */
@Entity
@Table(name = "learning_reports")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class LearningReport {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(name = "user_id", nullable = false)
  private Long userId;

  @Column(name = "period_from", nullable = false)
  private Instant periodFrom;

  @Column(name = "period_to", nullable = false)
  private Instant periodTo;

  @Column(nullable = false)
  private String status;

  // 生成完了後の成果物の S3 キー(未完了の間は null)。
  @Column(name = "s3_key")
  private String s3Key;

  @Column(name = "created_at")
  private Instant createdAt;
}

File: backend/internal/domain/learning_report.go
package domain

import "time"

// LearningReport はユーザーの学習レポート (週次・月次集計を非同期で生成)。
type LearningReport struct {
	ID         uint64    `json:"id"`
	UserID     uint64    `json:"userId"`
	PeriodFrom time.Time `json:"periodFrom"`
	PeriodTo   time.Time `json:"periodTo"`
	Status     string    `json:"status"`
	S3Key      string    `json:"s3Key,omitempty"`
	CreatedAt  time.Time `json:"createdAt"`
}

const (
	LearningReportStatusPending = "pending"
	LearningReportStatusReady   = "ready"
	LearningReportStatusFailed  = "failed"
)
